frame/collective/src/lib.rs

```diff
@@ -694,29 +551,236 @@ decl_module! {
+		WrongIndex,
+		/// Duplicate vote ignored
+		DuplicateVote,
+		/// Members are already initialized!
+		AlreadyInitialized,
+		/// The close call was made too early, before the end of the voting.
+		TooEarly,
+		/// There can only be a maximum of `MaxProposals` active proposals.
+		TooManyProposals,
+		/// The given weight bound for the proposal was too low.
+		WrongProposalWeight,
+		/// The given length bound for the proposal was too low.
+		WrongProposalLength,
+	}
+
+	/// The hashes of the active proposals.
+	#[pallet::storage]
+	#[pallet::getter(fn proposals)]
+	pub type Proposals<T: Config<I>, I: 'static = ()> = StorageValue<_, Vec<T::Hash>, ValueQuery>;
+
+	/// Actual proposal for a given hash, if it's current.
+	#[pallet::storage]
+	#[pallet::getter(fn proposal_of)]
+	pub type ProposalOf<T: Config<I>, I: 'static = ()> = StorageMap<
+		_,
+		Identity,
+		T::Hash,
+		T::Proposal,
+		OptionQuery,
+	>;
+
+	/// Votes on a given proposal, if it is ongoing.
+	#[pallet::storage]
+	#[pallet::getter(fn voting)]
+	pub type Voting<T: Config<I>, I: 'static = ()> = StorageMap<
+		_,
+		Identity,
+		T::Hash,
+		Votes<T::AccountId, T::BlockNumber>,
+		OptionQuery
+	>;
+
+	/// Proposals so far.
+	#[pallet::storage]
+	#[pallet::getter(fn proposal_count)]
+	pub type ProposalCount<T: Config<I>, I: 'static = ()> = StorageValue<_, u32, ValueQuery>;
+
+	/// The current members of the collective. This is stored sorted (just by value).
+	#[pallet::storage]
+	#[pallet::getter(fn members)]
+	pub type Members<T: Config<I>, I: 'static = ()> = StorageValue<_, Vec<T::AccountId>, ValueQuery>;
+
+	/// The prime member that helps determine the default vote behavior in case of absentations.
+	#[pallet::storage]
+	#[pallet::getter(fn prime)]
+	pub type Prime<T: Config<I>, I: 'static = ()> = StorageValue<_, T::AccountId, OptionQuery>;
+
+	/// (default) if not.
+	#[pallet::storage]
+	#[pallet::getter(fn upgraded_pallet_prefix)]
+	pub(super) type UpgradedPalletPrefix<T: Config<I>, I: 'static = ()> =
+		StorageValue<_, bool, ValueQuery>;
+
+	#[pallet::genesis_config]
+	pub struct GenesisConfig<T: Config<I>, I: 'static = ()>{
+		pub members: Vec<T::AccountId>,
+		pub phantom: sp_std::marker::PhantomData<(T, I)>,
+	}
+
+	#[cfg(feature = "std")]
+	impl<T: Config<I>, I: 'static> Default for GenesisConfig<T, I> {
+		fn default() -> Self {
+			Self {
+				members: Default::default(),
+				phantom: Default::default(),
+			}
+		}
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config<I>, I: 'static> GenesisBuild<T, I> for GenesisConfig<T, I> {
+		fn build(&self) {
+			<UpgradedPalletPrefix<T, I>>::put(true);
+			Pallet::<T, I>::initialize_members(&self.members);
+		}
+	}
+}
+
+mod migrations {
+	use super::*;
+
+	pub fn migrate_to_new_pallet_prefix<T: Config<I>, I: 'static>() -> frame_support::weights::Weight {
+
+		let new_name = T::PalletInfo::name::<Pallet<T, I>>()
+			.expect("Fatal Error Invalid PalletInfo name")
+			.as_bytes();
+
+		if new_name == b"Collective" {
+			return 0
+		}
```

Comment:
ah but this is wrong, the old prefix is `Collective` only for the default instance of the pallet.

For the instance 1 the prefix is `Instance1Collective` or something like this.
There is no way to know the old pallet prefix as `I` no longer bounds `Instance` trait.
```suggestion
```